Add preadd functionality

aadibajpai opened the pull request:
also fixed old react warnings about table elements not having unique keys

aadibajpai commented:
pls qa 😩 

I think editing card might be nice but not sure how to implement and might be nice to have a status icon signifying if the user who was preadded had logged in or something

aadibajpai commented on pages/api/preadd.ts:
```diff
@@ -0,0 +1,34 @@
+		case 'DELETE':
+			const { userId } = req.body;
+			console.log(`removing user with id ${userId} from preadded`);
+			const deleted = await PreAdd.findByIdAndDelete(userId);
```
I think it's fine bc we check if the person is organizer or not

aadibajpai commented:
also what is somewhat inconsistent I saw is that some components have their api handlers defined in the same file while some have them defined in OrganizerDash and passed down as props.

nkalupahana commented:
I can't pre-add on the preview? Getting a 500 whenever I submit. Also, just at a glance, it would be good if we could make sure the cards are the same size.

Ace314159 commented:
I can add just fine, but it does fail when you add duplicate entries. And yeah, I also agree the cards should be the same size.

nkalupahana commented:
Oh, you know what, I think I just tested with a duplicate instead of testing normally first. Yeah, that'd do it.

nkalupahana commented:
When you submit it says "Stored preadd information for undefined users" lol

aadibajpai commented:
> When you submit it says "Stored preadd information for undefined users" lol

okay should be fine now, I changed it to use the returned object. Also added handling of an empty submit.
